Bugsnag/Helpers/BSGRunContext.h

```diff
@@ -0,0 +1,24 @@
+//
+//  BSGRunContext.h
+//  Bugsnag
+//
+//  Copyright © 2022 Bugsnag Inc. All rights reserved.
+//
+
+#define BSGRUNCONTEXT_VERSION 1
+
+struct BSGRunContext {
+    long structVersion;
+};
+
+/// Information about the current run of the app / process.
+///
+/// This structure is mapped to a file so that changes will be persisted by the OS.
+///
+/// Guaranteed to be non-null once BSGRunContextInit() is called.
+extern struct BSGRunContext *_Nonnull bsg_runContext;
+
+/// Information about the last run of the app / process, if it could be loaded.
+extern const struct BSGRunContext *_Nullable bsg_lastRunContext;
```

Comment:
Do we have a way to ensure that the data loaded from disk is valid / safe to use? For example:

* App crashes immediately after memmapping - Does the file contain random data?
* App crashes immediately after memmapping, clearing, setting version - Is zeroed data safe to read from?

Reply:
When the file is created, `ftruncate()` fills it with zeros. It will never contain random data.

`structVersion` will be set _last_ to prevent partially initialised data from being read.